diff --git a/bfd/config.bfd b/bfd/config.bfd
--- a/bfd/config.bfd
+++ b/bfd/config.bfd
@@ -410,7 +410,7 @@ case "${targ}" in
     ;;
   arm-*-phoenix*)
     targ_defvec=arm_elf32_le_vec
-    targ_selvecs=arm_elf32_be_vec
+    targ_selvecs="arm_elf32_be_vec arm_elf32_fdpic_le_vec arm_elf32_fdpic_be_vec"
     ;;
   armeb-*-elf | arm*b-*-freebsd* | arm*b-*-linux-* | armeb-*-eabi*)
     targ_defvec=arm_elf32_be_vec
diff --git a/ld/Makefile.am b/ld/Makefile.am
--- a/ld/Makefile.am
+++ b/ld/Makefile.am
@@ -174,6 +174,7 @@ ALL_EMULATION_SOURCES = \
 	earmelf_nacl.c \
 	earmelf_nbsd.c \
 	earmelf_phoenix.c \
+	earmelf_phoenix_fdpiceabi.c \
 	earmelf_vxworks.c \
 	earmelfb.c \
 	earmelfb_fbsd.c \
diff --git a/ld/Makefile.in b/ld/Makefile.in
--- a/ld/Makefile.in
+++ b/ld/Makefile.in
@@ -685,6 +685,7 @@ ALL_EMULATION_SOURCES = \
 	earmelf_nacl.c \
 	earmelf_nbsd.c \
 	earmelf_phoenix.c \
+	earmelf_phoenix_fdpiceabi.c \
 	earmelf_vxworks.c \
 	earmelfb.c \
 	earmelfb_fbsd.c \
@@ -1311,6 +1312,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nacl.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nbsd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_phoenix.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_phoenix_fdpiceabi.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_vxworks.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelfb.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelfb_fbsd.Po@am__quote@
diff --git a/ld/configure.tgt b/ld/configure.tgt
--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -200,7 +200,7 @@ arm-*-nto*)		targ_emul=armnto
 			;;
 arm-*-phoenix*)
 			targ_emul=armelf_phoenix
-			targ_extra_emuls="armelf"
+			targ_extra_emuls="armelf armelf_phoenix_fdpiceabi"
 			targ_extra_libpath=$targ_extra_emuls
 			;;
 armeb-*-elf | armeb-*-eabi*)
diff --git a/ld/emulparams/armelf_phoenix_fdpiceabi.sh b/ld/emulparams/armelf_phoenix_fdpiceabi.sh
--- /dev/null
+++ b/ld/emulparams/armelf_phoenix_fdpiceabi.sh
@@ -0,0 +1,12 @@
+source_sh ${srcdir}/emulparams/armelf_phoenix.sh
+
+OUTPUT_FORMAT="elf32-littlearm-fdpic"
+BIG_OUTPUT_FORMAT="elf32-bigarm-fdpic"
+LITTLE_OUTPUT_FORMAT="elf32-littlearm-fdpic"
+
+OTHER_READONLY_SECTIONS="${OTHER_READONLY_SECTIONS}
+  .rofixup : {
+	${RELOCATING+__ROFIXUP_LIST__ = .;}
+	*(.rofixup)
+	${RELOCATING+__ROFIXUP_END__ = .;}
+}"
diff --git a/ld/po/BLD-POTFILES.in b/ld/po/BLD-POTFILES.in
--- a/ld/po/BLD-POTFILES.in
+++ b/ld/po/BLD-POTFILES.in
@@ -36,6 +36,7 @@ earmelf_linux_fdpiceabi.c
 earmelf_nacl.c
 earmelf_nbsd.c
 earmelf_phoenix.c
+earmelf_phoenix_fdpiceabi.c
 earmelf_vxworks.c
 earmelfb.c
 earmelfb_fbsd.c

