diff -Naur gcc-9.3.0-orig/libgcc/config/sparc/t-phoenix gcc-9.3.0/libgcc/config/sparc/t-phoenix
--- gcc-9.3.0-orig/libgcc/config/sparc/t-phoenix
+++ gcc-9.3.0/libgcc/config/sparc/t-phoenix
@@ -0,0 +1,5 @@
+# Phoenix-RTOS uses relocatable ELFs on NOMMU platforms
+# On SPARC without MMU -> build crtstuff with PIC
+ifneq (,$(findstring pic,$(CC)))
+	CRTSTUFF_T_CFLAGS = $(PICFLAG) -mcpu=leon3 -mno-pic-data-is-text-relative -mpic-register=g6
+endif
diff -Naur gcc-9.3.0-orig/libgcc/config.host gcc-9.3.0/libgcc/config.host
--- gcc-9.3.0-orig/libgcc/config.host
+++ gcc-9.3.0/libgcc/config.host
@@ -1316,5 +1316,6 @@ 
 sparc-*-phoenix*)
 	tmake_file="$tmake_file sparc/t-softmul t-fdpbit"
+	tmake_file="$tmake_file sparc/t-phoenix t-crtstuff-pic t-libgcc-pic"
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
 	;;
 sparc*-*-solaris2*)
