diff --git a/gcc/config.gcc b/gcc/config.gcc
index 95c91ee02be..7ad62278cfa 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1465,7 +1465,9 @@ arm*-*-phoenix*)
 	tm_file="elfos.h arm/unknown-elf.h arm/elf.h arm/bpabi.h"
 	tm_file="${tm_file} newlib-stdint.h phoenix.h"
 	tm_file="${tm_file} arm/aout.h arm/arm.h"
-	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-phoenix"
+	tmake_file="${tmake_file} arm/t-arm arm/t-bpabi arm/t-aprofile arm/t-rmprofile arm/t-multilib"
+	with_float="soft" # "auto" fpu
+	TM_MULTILIB_CONFIG="aprofile,rmprofile"
 	target_cpu_cname="arm7tdmi"
 	;;
 arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems* | arm*-*-fuchsia*)

