diff -Naur --exclude '*build*' --exclude '*gmp*' --exclude '*mpfr*' --exclude '*isl*' --exclude '*mpc*' gcc-9.3.0-orig/gcc/config.gcc gcc-9.3.0/gcc/config.gcc
--- gcc-9.3.0-orig/gcc/config.gcc	2020-10-19 13:24:43.512413024 +0200
+++ gcc-9.3.0/gcc/config.gcc	2020-10-19 13:41:49.274895804 +0200
@@ -2218,13 +2218,16 @@
 	# automatically detect that GAS supports it, yet we require it.
 	gcc_cv_initfini_array=yes
 	;;
-riscv*-*-elf* | riscv*-*-rtems*)
+riscv*-*-elf* | riscv*-*-rtems* | riscv*-*-phoenix*)
 	tm_file="elfos.h newlib-stdint.h ${tm_file} riscv/elf.h"
 	case ${target} in
 	*-*-rtems*)
 	  tm_file="${tm_file} rtems.h riscv/rtems.h"
 	  tmake_file="${tmake_file} riscv/t-rtems"
 	  ;;
+	*-*-phoenix*)
+	  tm_file="${tm_file} phoenix.h"
+	  ;;
 	*)
 	  case "x${enable_multilib}" in
 	  xno) ;;
