diff -Naur gcc-9.3.0-orig/libstdc++-v3/configure gcc-9.3.0/libstdc++-v3/configure
--- gcc-9.3.0-orig/libstdc++-v3/configure	2020-03-12 12:07:24.000000000 +0100
+++ cc-9.3.0/libstdc++-v3/configure	2023-02-24 13:44:47.346288244 +0100
@@ -899,6 +899,7 @@
 with_target_subdir
 with_cross_host
 with_newlib
+with_libphoenix
 enable_maintainer_mode
 enable_shared
 enable_static
@@ -1653,6 +1654,7 @@
                           configuring in a subdirectory
   --with-cross-host=HOST  configuring with a cross compiler
   --with-newlib           assume newlib as a system C library
+  --with-libphoenix       assume libphoenix as a system C library
   --with-pic              try to use only PIC/non-PIC objects [default=use
                           both]
   --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
@@ -5447,6 +5449,13 @@
 fi
 
 
+# Check whether --with-libphoenix was given.
+if test ${with_libphoenix+y}
+then :
+  withval=$with_libphoenix;
+fi
+
+
   # Will set LN_S to either 'ln -s', 'ln', or 'cp -p' (if linking isn't
   # available).  Uncomment the next line to force a particular method.
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether ln -s works" >&5
@@ -9068,6 +9077,31 @@
     lt_prog_compiler_static='-static'
 
     case $host_os in
+      phoenix)
+        case $host_cpu in
+          arm)
+            # Phoenix system for arms without MMU requires files to be compiled with some certain flags.
+            # lt_prog_compiler_pic is a variable which contains these flags for C files that are
+            # involved in libstdc++.
+
+            architecture=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
+
+            if [[ "$architecture" =~ "armv7-m" ]] || [[ "$architecture" =~ "armv7e-m" ]] || [[ "$architecture" =~ "armv8-m" ]]; then
+              lt_prog_compiler_pic="$lt_prog_compiler_pic -fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+            fi
+            ;;
+
+          sparc)
+
+            lt_prog_compiler_pic="$lt_prog_compiler_pic -fPIC -fPIE -mcpu=leon3 -mno-pic-data-is-text-relative -mpic-register=g6"
+            ;;
+
+          *)
+            lt_prog_compiler_pic='-fPIC'
+            ;;
+        esac
+        ;;
+
       aix*)
       # All AIX code is PIC.
       if test "$host_cpu" = ia64; then
@@ -9342,6 +9376,33 @@
   *djgpp*)
     lt_prog_compiler_pic=
     ;;
+
+  # Not all arm targets support PIC configuration
+  phoenix)
+    case $host_cpu in
+      arm)
+        # Phoenix system for arms without MMU requires files to be compiled with some certain flags.
+        # One of them is -fPIC which is complemented by macro -fPID. In this "case" is decided whether
+        # to add this macro or not.
+
+        architecture=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
+
+        if [[ "$architecture" =~ "armv7-m" ]] || [[ "$architecture" =~ "armv7e-m" ]] || [[ "$architecture" =~ "armv8-m" ]]; then
+          lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        fi
+        ;;
+
+      sparc)
+        lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        ;;
+
+      # For other targets the behaviour should be default
+      *)
+        lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        ;;
+    esac
+    ;;
+
   *)
     lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
     ;;
@@ -11755,6 +11816,9 @@
   lt_cv_dlopen_libs=
 
   case $host_os in
+  phoenix*)
+    ;;
+  
   beos*)
     lt_cv_dlopen="load_add_on"
     lt_cv_dlopen_libs=
@@ -13951,6 +14015,32 @@
     lt_prog_compiler_static_CXX='-static'
 
     case $host_os in
+    phoenix)
+      case $host_cpu in
+        arm)
+          # Phoenix system for arms without MMU requires files to be compiled with some certain flags.
+          # lt_prog_compiler_pic_CXX is a variable which contains these flags for C++ files that are
+          # involved in libstdc++.
+
+          architecture=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
+
+          if [[ "$architecture" =~ "armv7-m" ]] || [[ "$architecture" =~ "armv7e-m" ]] || [[ "$architecture" =~ "armv8-m" ]]; then
+            lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+          fi
+
+        ;;
+
+        sparc)
+
+          lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -fPIC -fPIE -mcpu=leon3 -mno-pic-data-is-text-relative -mpic-register=g6"
+          ;;
+
+        *)
+          lt_prog_compiler_pic_CXX='-fPIC'
+        ;;
+      esac
+    ;;
+
     aix*)
       # All AIX code is PIC.
       if test "$host_cpu" = ia64; then
@@ -14261,6 +14351,34 @@
   *djgpp*)
     lt_prog_compiler_pic_CXX=
     ;;
+
+  # Not all arm targets support PIC
+  phoenix)
+    case $host_cpu in
+      arm)
+        # Phoenix system for arms without MMU requires files to be compiled with some certain flags.
+        # One of them is -fPIC which is complemented by macro -fPID. In this "case" is decided whether
+        # to add this macro or not.
+
+        architecture=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
+
+        if [[ "$architecture" =~ "armv7-m" ]] || [[ "$architecture" =~ "armv7e-m" ]] || [[ "$architecture" =~ "armv8-m" ]]; then
+          lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
+        fi
+        ;;
+
+      sparc)
+
+        lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
+        ;;
+
+      # For other arm targets the behaviour should be default
+      *)
+        lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        ;;
+    esac
+    ;;
+
   *)
     lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
     ;;
@@ -15464,6 +15582,13 @@
   glibcxx_compiler_pic_flag="$lt_prog_compiler_pic_CXX"
   glibcxx_compiler_shared_flag="-D_GLIBCXX_SHARED"
 
+elif test "$pic_mode" = yes; then
+  # It was not possible to use pic without shared libraries in the default configure file
+  # due to the above code. If the "shaded" option is not enabled, the selected 
+  # lt_prog_compiler_pic_CXX flags will not be propagated to makefiles.
+
+  glibcxx_lt_pic_flag="-prefer-pic"
+  glibcxx_compiler_pic_flag="$lt_prog_compiler_pic_CXX"
 else
   glibcxx_lt_pic_flag=
   glibcxx_compiler_pic_flag=
@@ -29477,7 +29602,7 @@
   # Construct crosses by hand, eliminating bits that need ld...
   # GLIBCXX_CHECK_MATH_SUPPORT
 
-  # First, test for "known" system libraries.  We may be using newlib even
+  # First, test for "known" system libraries.  We may be using newlib or libphoenix even
   # on a hosted environment.
   if test "x${with_newlib}" = "xyes"; then
     os_include_dir="os/newlib"
@@ -29537,6 +29662,61 @@
 
     $as_echo "#define HAVE_MEMALIGN 1" >>confdefs.h
 
+  elif test "x${with_libphoenix}" = "xyes"; then
+    os_include_dir="os/generic"
+    printf "%s\n" "#define HAVE_HYPOT 1" >>confdefs.h
+
+
+    # GLIBCXX_CHECK_STDLIB_SUPPORT
+    printf "%s\n" "#define HAVE_STRTOF 1" >>confdefs.h
+
+
+    printf "%s\n" "#define HAVE_ACOSF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_ASINF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_ATAN2F 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_ATANF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_CEILF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_COSF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_COSHF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_EXPF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_FABSF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_FLOORF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_FMODF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_FREXPF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_LDEXPF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_LOG10F 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_LOGF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_MODFF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_POWF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_SINF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_SINHF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_SQRTF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_TANF 1" >>confdefs.h
+
+    printf "%s\n" "#define HAVE_TANHF 1" >>confdefs.h
+    
+    printf "%s\n" "#define HAVE_STRTOLD 1" >>confdefs.h
+
   else
 
 # Base decisions on target environment.
