diff --git a/libgcc/config/arm/t-phoenix b/libgcc/config/arm/t-phoenix
--- a/libgcc/config/arm/t-phoenix
+++ b/libgcc/config/arm/t-phoenix
@@ -8,6 +8,6 @@ PS_IS_CORTEX_M := $(findstring -m,$(PS_MARCH))
 PS_IS_CORTEX_R := $(findstring -r,$(PS_MARCH))
 
 ifneq ($(PS_IS_CORTEX_M)$(PS_IS_CORTEX_R),)
-  CRTSTUFF_T_CFLAGS = $(PICFLAG) -mno-pic-data-is-text-relative
-  INTERNAL_CFLAGS += $(PICFLAG) -mno-pic-data-is-text-relative
+  CRTSTUFF_T_CFLAGS = $(PICFLAG) -mfdpic -Wa,--fdpic -Wl,-marmelf_phoenix_fdpiceabi
+  INTERNAL_CFLAGS += $(PICFLAG) -mfdpic -Wa,--fdpic -Wl,-marmelf_phoenix_fdpiceabi
 endif
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -9249,7 +9249,7 @@ $as_echo_n "checking for $compiler option to produce PIC... " >&6; }
             arch=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
 
             if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
-              lt_prog_compiler_pic="-fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+              lt_prog_compiler_pic="-fPIC -fPIE -mfdpic -Wa,--fdpic -Wl,-marmelf_phoenix_fdpiceabi"
             else
               lt_prog_compiler_pic='-fPIC'
             fi
@@ -14307,7 +14307,7 @@ $as_echo_n "checking for $compiler option to produce PIC... " >&6; }
           arch=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
 
           if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
-            lt_prog_compiler_pic_CXX="-fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+            lt_prog_compiler_pic_CXX="-fPIC -fPIE -mfdpic -Wa,--fdpic -Wl,-marmelf_phoenix_fdpiceabi"
           else
             lt_prog_compiler_pic_CXX='-fPIC'
           fi

