diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index 4956052b6cd..9d7c9b34e40 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -9089,12 +9089,15 @@ $as_echo_n "checking for $compiler option to produce PIC... " >&6; }
             arch=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
 
             if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
-              lt_prog_compiler_pic="$lt_prog_compiler_pic -fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+              lt_prog_compiler_pic="-fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+            else
+              lt_prog_compiler_pic='-fPIC'
             fi
             ;;
 
           sparc)
             # Necessary flags are passed through multilib options.
+            lt_prog_compiler_pic='-fPIC'
             ;;
 
           *)
@@ -9390,6 +9393,8 @@ case $host_os in
 
         if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
           lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        else
+          lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
         fi
         ;;
 
@@ -11473,7 +11478,7 @@ fi
   dynamic_linker='GNU/Linux ld.so'
   ;;
 
-netbsd*)
+netbsd* | phoenix*)
   version_type=sunos
   need_lib_prefix=no
   need_version=no
@@ -14029,13 +14034,16 @@ $as_echo_n "checking for $compiler option to produce PIC... " >&6; }
           arch=$(echo "$CXX" | sed 's/.* -march=\([^ ]*\).*/\1/')
 
           if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
-            lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+            lt_prog_compiler_pic_CXX="-fPIC -fPIE -msingle-pic-base -mno-pic-data-is-text-relative"
+          else
+            lt_prog_compiler_pic_CXX='-fPIC'
           fi
 
         ;;
 
         sparc)
           # Necessary flags are passed through multilib options.
+          lt_prog_compiler_pic_CXX="-fPIC"
           ;;
 
         *)
@@ -14367,6 +14375,8 @@ case $host_os in
 
         if [[ "$arch" =~ "armv7-m" ]] || [[ "$arch" =~ "armv7-r" ]] || [[ "$arch" =~ "armv7e-m" ]] || [[ "$arch" =~ "armv8-m" ]] || [[ "$arch" =~ "armv8-r" ]]; then
           lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
+        else
+          lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
         fi
         ;;
 
@@ -14379,7 +14389,7 @@ case $host_os in
 
       # For other arm targets the behaviour should be default
       *)
-        lt_prog_compiler_pic="$lt_prog_compiler_pic -DPIC"
+        lt_prog_compiler_pic_CXX="$lt_prog_compiler_pic_CXX -DPIC"
         ;;
     esac
     ;;
@@ -15215,7 +15225,7 @@ fi
   dynamic_linker='GNU/Linux ld.so'
   ;;
 
-netbsd*)
+netbsd* | phoenix*)
   version_type=sunos
   need_lib_prefix=no
   need_version=no
@@ -29091,6 +29101,8 @@ else
     
     printf "%s\n" "#define HAVE_STRTOLD 1" >>confdefs.h
 
+    printf "%s\n" "#define HAVE___CXA_THREAD_ATEXIT_IMPL 1" >>confdefs.h
+
   else
 
 # Base decisions on target environment.
diff --git a/libstdc++-v3/configure.host b/libstdc++-v3/configure.host
index f2ff1295d00..aaf0eb2555b 100644
--- a/libstdc++-v3/configure.host
+++ b/libstdc++-v3/configure.host
@@ -366,7 +366,7 @@ case "${host}" in
 	;;
     esac
     ;;
-  arm*-*-freebsd*)
+  arm*-*-freebsd* | arm*-*-phoenix*)
      port_specific_symbol_files="\$(srcdir)/../config/os/gnu-linux/arm-eabi-extra.ver"
      ;;
   powerpc*-*-darwin*)
