diff --git a/gcc/config/phoenix.h b/gcc/config/phoenix.h
--- a/gcc/config/phoenix.h
+++ b/gcc/config/phoenix.h
@@ -26,7 +26,9 @@ along with GCC; see the file COPYING3.  If not see
       builtin_assert ("system=unix");      \
     } while (0)
 
-#define STD_LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
+/* use gcc default for libc (re-define as it might have been overridden by previous imports) */
+#undef LIB_SPEC
+#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
 
 /* This will prevent selecting 'unsigned long int' instead of 'unsigned int' as 'uint32_t' in stdint-newlib.h. */
 #undef STDINT_LONG32

