diff -Naur gcc-9.3.0-orig/gcc/config/sparc/t-phoenix gcc-9.3.0/gcc/config/sparc/t-phoenix
--- gcc-9.3.0-orig/gcc/config/sparc/t-phoenix
+++ gcc-9.3.0/gcc/config/sparc/t-phoenix
@@ -0,0 +1,14 @@
+MULTILIB_OPTIONS = mcpu=leon3 msoft-float mfix-gr712rc fPIE mno-pic-data-is-text-relative mpic-register=g6
+MULTILIB_DIRNAMES = leon3 soft gr712rc pie no-pditr pic-reg-g6
+MULTILIB_MATCHES = msoft-float=mno-fpu fPIE=fpie
+
+MULTILIB_REQUIRED =
+# GR712RC
+MULTILIB_REQUIRED += mcpu=leon3/mfix-gr712rc
+MULTILIB_REQUIRED += mcpu=leon3/msoft-float/mfix-gr712rc
+# LEON3 no MMU (pic/pie) - GR716 apps
+MULTILIB_REQUIRED += mcpu=leon3/fPIE/mno-pic-data-is-text-relative/mpic-register=g6
+MULTILIB_REQUIRED += mcpu=leon3/msoft-float/fPIE/mno-pic-data-is-text-relative/mpic-register=g6
+# LEON3 no pic/pie
+MULTILIB_REQUIRED += mcpu=leon3
+MULTILIB_REQUIRED += mcpu=leon3/msoft-float
diff -Naur gcc-9.3.0-orig/gcc/config.gcc gcc-9.3.0/gcc/config.gcc
--- gcc-9.3.0-orig/gcc/config.gcc	2020-03-12 12:07:21.000000000 +0100
+++ gcc-9.3.0/gcc/config.gcc	2020-03-18 19:25:57.937836171 +0100
@@ -3028,6 +3028,10 @@
 sparc-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h sparc/sysv4.h sparc/sp-elf.h sparc/rtemself.h rtems.h newlib-stdint.h"
 	tmake_file="${tmake_file} sparc/t-sparc sparc/t-rtems"
 	;;
+sparc-*-phoenix*)
+	tm_file="${tm_file} dbxelf.h elfos.h sparc/sysv4.h sparc/sp-elf.h phoenix.h newlib-stdint.h"
+	tmake_file="${tmake_file} sparc/t-sparc sparc/t-phoenix"
+	;;
 sparc-*-linux*)
 	tm_file="${tm_file} dbxelf.h elfos.h sparc/sysv4.h gnu-user.h linux.h glibc-stdint.h sparc/tso.h"
diff -Naur gcc-9.3.0-orig/libgcc/config.host gcc-9.3.0/libgcc/config.host
--- gcc-9.3.0-orig/libgcc/config.host	2020-03-12 12:07:23.000000000 +0100
+++ gcc-9.3.0/libgcc/config.host	2020-03-18 15:31:59.377075283 +0100
@@ -1312,7 +1312,11 @@
 sparc-*-rtems*)
 	tmake_file="$tmake_file sparc/t-elf sparc/t-softmul t-crtfm t-fdpbit"
 	extra_parts="$extra_parts crti.o crtn.o crtfastmath.o"
 	;;
+sparc-*-phoenix*)
+	tmake_file="$tmake_file sparc/t-softmul t-fdpbit"
+	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
+	;;
 sparc*-*-solaris2*)
 	tmake_file="$tmake_file t-crtfm"
 	extra_parts="$extra_parts crtfastmath.o"
