diff --git a/gcc/config/phoenix.h b/gcc/config/phoenix.h
--- a/gcc/config/phoenix.h
+++ b/gcc/config/phoenix.h
@@ -31,3 +31,5 @@ along with GCC; see the file COPYING3.  If not see
 /* This will prevent selecting 'unsigned long int' instead of 'unsigned int' as 'uint32_t' in stdint-newlib.h. */
 #undef STDINT_LONG32
 #define STDINT_LONG32		0
+/* libitm is not supported on Phoenix - disable it in crtstuff */
+#define USE_TM_CLONE_REGISTRY 0

